testing/chain.go

```diff
@@ -261,13 +262,18 @@ func (chain *TestChain) QueryConsensusStateProof(clientID string) ([]byte, clien
 
 // NextBlock sets the last header to the current header and increments the current header to be
 // at the next block height. It does not update the time as that is handled by the Coordinator.
-//
-// CONTRACT: this function must only be called after app.Commit() occurs
 func (chain *TestChain) NextBlock() {
+	res := chain.App.EndBlock(abci.RequestEndBlock{Height: chain.CurrentHeader.Height})
+
+	chain.App.Commit()
```

Comment:
EndBlock and Commit are now called within `NextBlock`. This allows us to correctly track validator changes as seen below

Reply:
Can you add this information to the godoc?